instruqt/protecting-your-assets/04-helm-values-for-license-fields/assignment.md

```diff
@@ -0,0 +1,273 @@
+trial.
+
+Using the editor in the "Slackernews Chart" tab, open the file
+`slackernews/templates/slackernews-deployment`.
+
+![Editing the Slackernews Deployment](../assets/editing-the-deployment.png)
+
+We're going to put a guard condition around the environment variables that
+configure the database so that customers using a trial license can not use a
+Postgres database for long-term persistence. There's already a check to see if
+Postgres is enabled, so let's modify that using a helper template.
+
+Find the line that says `{{ if eq .Values.postgres.enabled true }}` and
+replace it with the following line
+
+```
+{{ if eq ( include "slackernews.persistence.enabled" .) "true" }}
+```
+
+The template we're including isn't defined, so let's add it to the
+`_helpers.tpl` file with other helper templates. Open that file (also under
+`slackernews/templates` and add the definition at the end.
+
+```
+{{- define "slackernews.persistence.enabled" -}}
+{{- $licenseType := "prod" -}}
+{{- $postgresEnabled := .Values.postgres.enabled -}}
+{{- if hasKey .Values "global" -}}
+  {{- if hasKey .Values.global "replicated" -}}
+    {{- if hasKey .Values.global.replicated "licenseType" -}}
+      {{- $licenseType = .Values.global.replicated.licenseType -}}
+    {{- end -}}
+  {{- end -}}
+{{- end -}}
+{{- and (ne $licenseType "trial") $postgresEnabled -}}
+{{- end }}
+```
+
+With these additions, the template will not allow the use of the Postgres
+database if the license is a trial license.
+
+Trying it Out
+=============
+
+We've already configured Nitflex with a trial license, so let's try out our
+changes. First we'll have to release them, then we'll try an install and see
+what happens.
+
+### Releasing the Updates Chart
+
+First bump the version of the Helm chart. You can do it in the editor by
+updating `Chart.yaml`, or use the following command.
+
+```
+yq -i '.version = "0.5.1"' slackernews/Chart.yaml
+```
+
+Package up the new version and release it to the three default release
+channels.
+
+```
+helm package -u slackernews -d release
+replicated release create --promote Unstable --chart release/slackernews-0.5.1.tgz --version 0.5.1  \
+  --release-notes "Prevents persistence with trial licenses"
+replicated release promote 6 Beta --version 0.5.1 \
+  --release-notes "Prevents persistence with trial licenses"
+replicated release promote 6 Stable --version 0.5.1 \
+  --release-notes "Prevents persistence with trial licenses"
+```
+
+### Installing
+
+Let's install the latest release as Nitflex, then check to see whether
+persistence with Postgres was enabled. Since it's a good practice, let's make
+sure we run our preflight checks before installing.
+
+```
+helm template oci://registry.replicated.com/[[ Instruqt-Var key="REPLICATED_APP" hostname="shell" ]]/slackernews \
+  | kubectl preflight -
+```
+
+Your preflight checks should pass, so let's install. Postgres is enabled by the
```

Comment:
![image](https://github.com/replicatedhq/replicated-field-labs/assets/5035609/770407aa-3c29-4819-a690-21c688afd238)

preflights aren't happy (not that it's a problem...)

Reply:
I thought I bumped up the machine size, I'll have to do that.

Reply:
Did it in a previous lab but not this one. Bumped it up now.